client/src/wm.c

```diff
@@ -155,3 +191,163 @@ void wmWarpMouse(int x, int y)
       break;
   }
 }
+
+// Wayland support.
+
+struct WMDataWayland
+{
+  struct wl_display * display;
+  struct wl_registry * registry;
+  struct wl_seat * seat;
+
+  uint32_t capabilities;
+
+  struct wl_pointer * pointer;
+  struct zwp_relative_pointer_manager_v1 * relativePointerManager;
+  struct zwp_relative_pointer_v1 * relativePointer;
+  struct zwp_locked_pointer_v1 * lockedPointer;
+
+  double dx_frac;
+  double dy_frac;
+};
+
+// Registry-handling listeners.
+
+static void registryGlobalHandler(void * data, struct wl_registry * registry,
+    uint32_t name, const char * interface, uint32_t version)
+{
+  struct WMDataWayland * wm = data;
+
+  fprintf(stderr, "%s\n", interface);
```

Comment:
More debug output?
`fprintf` and friends should never be used directly in the LG project, please use `DEBUG_INFO`, `DEBUG_WARN` or `DEBUG_ERROR`.